Source/Orts.Simulation/Simulation/AIs/AIPath.cs

```diff
@@ -516,7 +509,7 @@ public static int FindJunctionOrEndIndex(WorldLocation location, TrackDB trackDB
                 float dz = tn.UiD.Z - location.Location.Z;
                 dz += (tn.UiD.TileZ - location.TileZ) * 2048;
                 float dy = tn.UiD.Y - location.Location.Y;
-                float d = dx * dx + dy * dy + dz * dz;
+                float d = (dx * dx) + (dy * dy) + (dz * dz);
```

Comment:
Are this kind of unneeded parenthesis addition going to be enforced? It may be because I come from the Math/Physics fields instead of Computer Science, but I find the new expression more difficult to read. It's not a common thing there to put parenthesis around multiplications, everyone knows that they go first. When I see a parenthesis, my mind thinks "there is a different operator precedence than usual", to then see (in this situation) that the operator precedence is the standard one, effectively loosing more time than required when analyzing a formula.

To me, adding parenthesis to
```csharp
scalar = (dx*dx) + (dy*dy)
```
is the same as
```csharp
float a = (3+5);
```
i.e. both add unnecessary parenthesis that make the code harder to read. Why is it assumed that you know = goes after +, but not that addition goes after multiplication? To me this is completely inconsistent, and makes the code harder to read. My opinion is don't add parenthesis to mathematical operations unless they are required.

I'll still use parenthesis only when needed, unless this new rule is enforced. I would be very happy also if equations used for Physics calculations are kept as they are. Some of them are very complex, and having to waste time to match unnecessary parenthesis does no help.

Maybe there's a good reason to flood the code with parenthesis, but at the moment I can only see its drawbacks.

Reply:
We have had a default set of style rules in the repository for some time, with the goal of slowing bringing all the code style back into a consistent shape. This is, I think, the first large-scale formatting to occur that could be accepted (myself and others have trialled formatting the whole repo, but the changes were unmanageable).

Nothing enforces any of the rules.

Visual Studio will apply the rules to snippets, completions, refactors, and other places where it generates code.

The relevant rules here are all set to the defaults and have not been customised. That means that this is both implicitly the recommended style and is, or will become, the most common style of C# code.

I am generally open to changing the rules if there are strong reasons, and you can see our customisations (which match existing Open Rails style) here: https://github.com/openrails/openrails/blob/d3944727386daae3fa7fdce2910280479865ac0b/Source/.editorconfig#L137-L160

However, I don't find "everyone knows" a convincing argument for change. And while "difficult to read" is a better argument, it is subjective. I find the parentheses beneficial, for example.

The parenthesis rules are here: https://github.com/openrails/openrails/blob/d3944727386daae3fa7fdce2910280479865ac0b/Source/.editorconfig#L38-L42

The defaults are from here: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-style-rule-options#example-editorconfig-file

Reply:
Thanks for your message. I'm still not convinced about it (I don't think "Microsoft sets this as the default" is a strong enough argument to overrule standard math conventions), but I'll bear with it. It's good to have some global consistency in the code.

My opinion is that this shouldn't be automatically refactored, but leave it to the coder's choice. The original author may add parenthesis if he thinks they are necessary for clarity. But as you say this is subjective.